### DISCUSSION TITLE
Deleting record from store breaks array.map of array that contained deleted record

### DISCUSSION THREAD

**Discussion by axhl** [sibelius/relay-workshop-old#7] [Q&A] [answered]
I have a relay fragment which returns, amongst other things, an array which I map over in a basic list (simplified example):

`fragmentArrayData.map((x) => {<li>{x.property}</li>})
`

onClick, each item in the list opens a modal, from which the array item can be deleted with a basic mutation-hook. The mutation has an `updater` which deletes the item from the relay store to prevent having to refresh the page to see the update: `store.delete(x.id)`

Rather than successfully re-run `fragmentArrayData.map`, my app tries to iterate over the now deleted item in the array, with `Cannot read property...` and as the breaking result. 

Any thoughts what I am missing? Grateful for input

**Comment by sibelius** [comment, marked as answer]
Relay won’t remove a null node from your array, so you need to remove it by yourself inside updater

Removing a node from a connection works better with Relay

Relay treats arrays more like immutable, or something that would also change together

**Reply by axhl** [reply]
Many thanks for jumping on this! Null node makes sense.
